Quarkus 1.3.0.CR2 Gradle multi module CDI UnsatisfiedResolutionException

**Describe the bug**
I have a gradle multi module setup. One module is used as a library, the other is the quarkus application using this library. The library use and includes the `KubernetesClient`. When I also add the smallrye-health extension to the library, the application can no longer inject the `KubernetesClient`. The same setup works with Quarkus 1.2.1.Final.

**Expected behavior**
KubernetesClient should be injected.

**Actual behavior**
Exception is thrown on quarkusDev start or quarkusBuild

```
2020-03-12 15:33:12,902 ERROR [io.qua.dev.DevModeMain] (main) Failed to start Quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: javax.enterprise.inject.spi.DeploymentException: Found 2 deployment problems: 
[1] Unsatisfied dependency for type io.fabric8.kubernetes.client.KubernetesClient and qualifiers [@Default]
	- java member: org.acme.ExampleResource#<init>()
	- declared on CLASS bean [types=[org.acme.ExampleResource, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.ExampleResource]
[2] Unsatisfied dependency for type io.fabric8.kubernetes.client.KubernetesClient and qualifiers [@Default]
	- java member: org.acme.BeanWithInjection#kubernetesClient
	- declared on CLASS bean [types=[org.acme.BeanWithInjection, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.BeanWithInjection]
	at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:918)
	at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:232)
	at io.quarkus.arc.processor.BeanProcessor.initialize(BeanProcessor.java:130)
	at io.quarkus.arc.deployment.ArcProcessor.validate(ArcProcessor.java:291)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:938)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:273)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2027)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1551)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1442)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
	Suppressed: javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type io.fabric8.kubernetes.client.KubernetesClient and qualifiers [@Default]
	- java member: org.acme.ExampleResource#<init>()
	- declared on CLASS bean [types=[org.acme.ExampleResource, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.ExampleResource]
		at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:472)
		at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:420)
		at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:224)
		... 14 more
	Suppressed: javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type io.fabric8.kubernetes.client.KubernetesClient and qualifiers [@Default]
	- java member: org.acme.BeanWithInjection#kubernetesClient
	- declared on CLASS bean [types=[org.acme.BeanWithInjection, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.BeanWithInjection]
		at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:472)
		at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:420)
		at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:224)
		... 14 more

	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:206)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:95)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:45)
	at io.quarkus.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:59)
	at io.quarkus.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:236)
	at io.quarkus.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:39)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:131)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:84)
	at io.quarkus.dev.DevModeMain.start(DevModeMain.java:108)
	at io.quarkus.dev.DevModeMain.main(DevModeMain.java:47)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: javax.enterprise.inject.spi.DeploymentException: Found 2 deployment problems: 
[1] Unsatisfied dependency for type io.fabric8.kubernetes.client.KubernetesClient and qualifiers [@Default]
	- java member: org.acme.ExampleResource#<init>()
	- declared on CLASS bean [types=[org.acme.ExampleResource, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.ExampleResource]
[2] Unsatisfied dependency for type io.fabric8.kubernetes.client.KubernetesClient and qualifiers [@Default]
	- java member: org.acme.BeanWithInjection#kubernetesClient
	- declared on CLASS bean [types=[org.acme.BeanWithInjection, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.BeanWithInjection]
	at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:918)
	at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:232)
	at io.quarkus.arc.processor.BeanProcessor.initialize(BeanProcessor.java:130)
	at io.quarkus.arc.deployment.ArcProcessor.validate(ArcProcessor.java:291)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:938)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:273)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2027)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1551)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1442)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
	Suppressed: javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type io.fabric8.kubernetes.client.KubernetesClient and qualifiers [@Default]
	- java member: org.acme.ExampleResource#<init>()
	- declared on CLASS bean [types=[org.acme.ExampleResource, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.ExampleResource]
		at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:472)
		at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:420)
		at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:224)
		... 14 more
	Suppressed: javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type io.fabric8.kubernetes.client.KubernetesClient and qualifiers [@Default]
	- java member: org.acme.BeanWithInjection#kubernetesClient
	- declared on CLASS bean [types=[org.acme.BeanWithInjection, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.BeanWithInjection]
		at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:472)
		at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:420)
		at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:224)
		... 14 more

	at io.quarkus.builder.Execution.run(Execution.java:115)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:143)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:204)
	... 9 more
Caused by: javax.enterprise.inject.spi.DeploymentException: Found 2 deployment problems: 
[1] Unsatisfied dependency for type io.fabric8.kubernetes.client.KubernetesClient and qualifiers [@Default]
	- java member: org.acme.ExampleResource#<init>()
	- declared on CLASS bean [types=[org.acme.ExampleResource, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.ExampleResource]
[2] Unsatisfied dependency for type io.fabric8.kubernetes.client.KubernetesClient and qualifiers [@Default]
	- java member: org.acme.BeanWithInjection#kubernetesClient
	- declared on CLASS bean [types=[org.acme.BeanWithInjection, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.BeanWithInjection]
	at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:918)
	at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:232)
	at io.quarkus.arc.processor.BeanProcessor.initialize(BeanProcessor.java:130)
	at io.quarkus.arc.deployment.ArcProcessor.validate(ArcProcessor.java:291)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:938)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:273)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2027)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1551)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1442)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
	Suppressed: javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type io.fabric8.kubernetes.client.KubernetesClient and qualifiers [@Default]
	- java member: org.acme.ExampleResource#<init>()
	- declared on CLASS bean [types=[org.acme.ExampleResource, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.ExampleResource]
		at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:472)
		at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:420)
		at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:224)
		... 14 more
	Suppressed: javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type io.fabric8.kubernetes.client.KubernetesClient and qualifiers [@Default]
	- java member: org.acme.BeanWithInjection#kubernetesClient
	- declared on CLASS bean [types=[org.acme.BeanWithInjection, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.BeanWithInjection]
		at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:472)
		at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:420)
		at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:224)
		... 14 more

2020-03-12 15:33:12,902 INFO  [io.qua.dev.DevModeMain] (main) Attempting to start hot replacement endpoint to recover from previous Quarkus startup failure
> Task :application:quarkusDev
```


**To Reproduce**
Steps to reproduce the behavior:
1. Download reproducer project 
[multi-module-transitive-not-working.zip](https://github.com/quarkusio/quarkus/files/4324707/multi-module-transitive-not-working.zip) and unzip it.
2. In the directory, run `./gradlew quarkusDev`
3. Exception is shown in log.


**Environment (please complete the following information):**
 - Quarkus 1.3.0.CR2 as well as master edec6a400b5a9b8b8bf35d9ce75f135de860e64c
 - works with Quarkus 1.2.1.Final
 - Windows 10

--- a/devtools/gradle/src/functionalTest/resources/basic-java-library-module/library/build.gradle
+++ b/devtools/gradle/src/functionalTest/resources/basic-java-library-module/library/build.gradle
@@ -10,6 +10,7 @@ repositories {
 dependencies {
     api enforcedPlatform("${quarkusPlatformGroupId}:${quarkusPlatformArtifactId}:${quarkusPlatformVersion}")
     api 'io.quarkus:quarkus-kubernetes-client'
+    api 'io.quarkus:quarkus-smallrye-health'
 
     testImplementation 'io.quarkus:quarkus-junit5'
     testImplementation 'io.rest-assured:rest-assured'

--- a/devtools/gradle/src/main/java/io/quarkus/gradle/AppModelGradleResolver.java
+++ b/devtools/gradle/src/main/java/io/quarkus/gradle/AppModelGradleResolver.java
@@ -286,13 +286,15 @@ public class AppModelGradleResolver implements AppModelResolver {
                 }
             }
 
+            boolean addChildExtensions = firstLevelExt;
             if (extDep != null && firstLevelExt) {
-                firstLevelExt = false;
                 firstLevelExtensions.add(extDep);
+                addChildExtensions = false;
             }
             final Set<ResolvedDependency> resolvedChildren = dep.getChildren();
             if (!resolvedChildren.isEmpty()) {
-                collectExtensionDeps(resolvedChildren, versionMap, appBuilder, firstLevelExtensions, firstLevelExt, visited);
+                collectExtensionDeps(resolvedChildren, versionMap, appBuilder, firstLevelExtensions, addChildExtensions,
+                        visited);
             }
         }
     }

